Marlin/src/feature/pause.cpp

```diff
@@ -252,94 +256,143 @@ void wait_for_filament_reload(const int8_t max_beep_count/*=0*/) {
-        filament_change_beep(max_beep_count);
+        if (swap_spool_disabled) filament_change_beep(max_beep_count);
       #endif
       idle(true);
     }
     KEEPALIVE_STATE(IN_HANDLER);
 
     #if ENABLED(ULTIPANEL)
       // Show "load" message
-      lcd_advanced_pause_show_message(ADVANCED_PAUSE_MESSAGE_LOAD);
+      if (swap_spool_disabled) lcd_advanced_pause_show_message(ADVANCED_PAUSE_MESSAGE_LOAD); //only disabled for automatic tool change on M600/runout//steeve
     #endif
 
     // Load filament
-    do_pause_e_move(load_length, FILAMENT_CHANGE_LOAD_FEEDRATE);
+    do_pause_e_move(load_length, PAUSE_LOAD_F);
   }
 
-  #if ENABLED(ULTIPANEL) && ADVANCED_PAUSE_EXTRUDE_LENGTH > 0
+  #if ENABLED(ULTIPANEL) && PAUSE_EXTRUDE > 0
 
     float extrude_length = initial_extrude_length;
 
-    do {
-      if (extrude_length > 0) {
-        // "Wait for filament extrude"
-        lcd_advanced_pause_show_message(ADVANCED_PAUSE_MESSAGE_EXTRUDE);
-
-        // Extrude filament to get into hotend
-        do_pause_e_move(extrude_length, ADVANCED_PAUSE_EXTRUDE_FEEDRATE);
-      }
-
-      // Show "Extrude More" / "Resume" menu and wait for reply
-      KEEPALIVE_STATE(PAUSED_FOR_USER);
-      wait_for_user = false;
-      lcd_advanced_pause_show_message(ADVANCED_PAUSE_MESSAGE_OPTION);
-      while (advanced_pause_menu_response == ADVANCED_PAUSE_RESPONSE_WAIT_FOR) idle(true);
-      KEEPALIVE_STATE(IN_HANDLER);
-
-      extrude_length = ADVANCED_PAUSE_EXTRUDE_LENGTH;
-
-      // Keep looping if "Extrude More" was selected
-    } while (advanced_pause_menu_response == ADVANCED_PAUSE_RESPONSE_EXTRUDE_MORE);
-
+    if (swap_spool_disabled){ 
+        
+	  do {
+	    if (extrude_length > 0) {
+		  // "Wait for filament extrude"
+		  if (swap_spool_disabled) lcd_advanced_pause_show_message(ADVANCED_PAUSE_MESSAGE_EXTRUDE); 
+		  // Extrude filament to get into hotend
+          do_pause_e_move(extrude_length, PAUSE_EXTRUDE_F);
+	    }
+		// Show "Extrude More" / "Resume" menu and wait for reply
+		KEEPALIVE_STATE(PAUSED_FOR_USER);
+		wait_for_user = false;
+		lcd_advanced_pause_show_message(ADVANCED_PAUSE_MESSAGE_OPTION);
+	    while (advanced_pause_menu_response == ADVANCED_PAUSE_RESPONSE_WAIT_FOR) idle(true);
+		KEEPALIVE_STATE(IN_HANDLER);
+		
+		extrude_length = PAUSE_EXTRUDE;
+		// Keep looping if "Extrude More" was selected
+	  } while (advanced_pause_menu_response == ADVANCED_PAUSE_RESPONSE_EXTRUDE_MORE);
+  }
+  else do_pause_e_move(extrude_length, PAUSE_EXTRUDE_F);
   #endif
-
+  
+  // Extruder purged = initialisation  
+  planner.set_e_position_mm(0); current_position[E_AXIS] = 0; destination[E_AXIS]=0;
+  
   #if ENABLED(ULTIPANEL)
     // "Wait for print to resume"
-    lcd_advanced_pause_show_message(ADVANCED_PAUSE_MESSAGE_RESUME);
+    if (swap_spool_disabled)lcd_advanced_pause_show_message(ADVANCED_PAUSE_MESSAGE_RESUME);
+  #endif
+	
+  // Stop blowing after delay
+  #ifdef PAUSE_DWELL
+    delay(PAUSE_DWELL);
+  #endif
```

Comment:
delay after extrusion/blowing
purged filament is hard , it cuts perfect , when return to print